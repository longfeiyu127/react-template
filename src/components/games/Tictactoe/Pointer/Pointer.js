import React from 'react';
// import PropTypes from 'prop-types';
import './Pointer.less';

export function slantType(line) {
  return line && line.every(item => item[0] === item[1]) ? 'slant0' : 'slant1';
}

export function checkLineType(line) {
  if (line) {
    const x = line[0][1];
    const y = line[0][0];
    if (line.every(item => x === item[1])) {
      return 'c';
    }
    if (line.every(item => y === item[0])) {
      return 'v';
    }
    return 's';
  }
  return line;
}

const TictactoePointer = props => {
  // console.log(props);
  // eslint-disable-next-line
  const { line } = props;
  const lineType = checkLineType(line);
  return (
    <div className={['g-tic-pointer', line ? '' : 'hide'].join(' ')}>
      <div
        className={[
          'pointer',
          'crosswise',
          lineType && lineType === 'c' ? `crosswise${line[0][1]}` : 'hide'
        ].join(' ')}
      />
      <div
        className={[
          'pointer',
          'vertical',
          lineType && lineType === 'v' ? `vertical${line[0][0]}` : 'hide'
        ].join(' ')}
      />
      <div
        className={[
          'pointer',
          'slant',
          lineType && lineType === 's' ? slantType(line) : 'hide'
        ].join(' ')}
      />
    </div>
  );
};

// TictactoePointer.propTypes = {
//   line: PropTypes.arrayof(PropTypes.oneOf([PropTypes.array, undefined])).isRequired
// };

export default TictactoePointer;
