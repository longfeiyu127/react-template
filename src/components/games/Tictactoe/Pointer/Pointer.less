.g-tic-pointer{
  position: absolute;
  top: 0;
  left: 0;
  height: 100%;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  .pointer{
    height: 0.13rem;
    width: 90%;
    background: url('../../../../assets//images//games/tictactoe/result.png') no-repeat center;
    background-size: 100% 100%;
  }
  .crosswise0{
    position: relative;
    top: -34%;
  }
  .crosswise2{
    position: relative;
    top: 32%;
  }
  .vertical{
    transform:rotate(90deg);
  }
  .vertical0{
    position: relative;
    left: -34%;
  }
  .vertical2{
    position: relative;
    left: 33%;
  }
  .slant{
    width: 100%;
  }
  .slant0{
    transform:rotate(45deg);
  }
  .slant1{
    transform:rotate(-45deg);
  }
}