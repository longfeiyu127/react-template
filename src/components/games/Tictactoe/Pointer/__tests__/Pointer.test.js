import React from 'react';
import renderer from 'react-test-renderer';
import Pointer, { slantType, checkLineType } from '../Pointer';

const line = [[0, 0], [1, 1], [2, 2]];
const line1 = [[0, 0], [0, 1], [0, 2]];
const line2 = [[0, 0], [1, 0], [2, 0]];

describe('<Pointer />', () => {
  it('component render', () => {
    let pointer = renderer.create(<Pointer line={line} />).toJSON();
    expect(pointer).toMatchSnapshot();

    pointer = renderer.create(<Pointer line={line1} />).toJSON();
    expect(pointer).toMatchSnapshot();

    pointer = renderer.create(<Pointer line={line2} />).toJSON();
    expect(pointer).toMatchSnapshot();

    pointer = renderer.create(<Pointer line={false} />).toJSON();
    expect(pointer).toMatchSnapshot();
  });

  it('function', () => {
    expect(slantType(line)).toBe('slant0');
    expect(slantType(line2)).toBe('slant1');
    expect(checkLineType(line)).toBe('s');
    expect(checkLineType(line1)).toBe('v');
    expect(checkLineType(line2)).toBe('c');
    expect(checkLineType(undefined)).toBeUndefined();
  });
});
