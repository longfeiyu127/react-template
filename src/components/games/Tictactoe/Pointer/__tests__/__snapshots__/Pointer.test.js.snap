// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pointer /> component render 1`] = `
<div
  className="g-tic-pointer "
>
  <div
    className="pointer crosswise hide"
  />
  <div
    className="pointer vertical hide"
  />
  <div
    className="pointer slant slant0"
  />
</div>
`;

exports[`<Pointer /> component render 2`] = `
<div
  className="g-tic-pointer "
>
  <div
    className="pointer crosswise hide"
  />
  <div
    className="pointer vertical vertical0"
  />
  <div
    className="pointer slant hide"
  />
</div>
`;

exports[`<Pointer /> component render 3`] = `
<div
  className="g-tic-pointer "
>
  <div
    className="pointer crosswise crosswise0"
  />
  <div
    className="pointer vertical hide"
  />
  <div
    className="pointer slant hide"
  />
</div>
`;

exports[`<Pointer /> component render 4`] = `
<div
  className="g-tic-pointer hide"
>
  <div
    className="pointer crosswise hide"
  />
  <div
    className="pointer vertical hide"
  />
  <div
    className="pointer slant hide"
  />
</div>
`;
